core/src/test/scala/zio/config/ProductBuilderTest.scala

```diff
@@ -27,8 +29,8 @@ object ProductBuilderTest
         },
         testM("combine 22 for tupled") {
           checkM(genS22) { p =>
-            val tuple =
-              S22.unapply(p).get
+            val conv = implicitly[TupleConversion[S22, (Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int)]]
```

Comment:
Should we add a new test case for TupleConversion and keep the old one as it is?

Reply:
If we do that it has to be moved to a Scala2-only source set as it won't compile on Scala3

Reply:
Right.. Nope this is fine.